src/screens/TopicPageScreen/MemoizedListComponent.js

```diff
@@ -121,6 +126,9 @@ const RenderListFeed = (props) => {
           hideThreeDot={true}
           props={item}
           height={getHeightHeader()}
+          isSelf={item?.is_self}
+          isFollow={item?.is_following_target}
+          onPressFollUnFoll={() => followUnfollowTopic(item, route?.params?.id, offset)}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onPressFollUnFoll` prop of the `Header` component now calls the `followUnfollowTopic` function with the item, the id parameter from the current route, and the `offset`. This change enables the follow/unfollow functionality for topics.
```diff
+           isSelf={item?.is_self}
+           isFollow={item?.is_following_target}
+           onPressFollUnFoll={() => followUnfollowTopic(item, route?.params?.id, offset)}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->